File: backend/src/main/java/com/banana/hotel/common/enums/RoomStatus.java
package com.banana.hotel.common.enums;

import lombok.Getter;

@Getter
public enum RoomStatus {

    CLEAN(0, "空净"),
    RESERVED(1, "预留"),
    BOOKED(2, "预订"),
    OCCUPIED(3, "在住"),
    DIRTY(4, "脏房"),
    OUT_OF_SERVICE(5, "停售");

    private final int code;
    private final String label;

    RoomStatus(int code, String label) {
        this.code = code;
        this.label = label;
    }

    public static RoomStatus fromCode(int code) {
        for (RoomStatus status : values()) {
            if (status.code == code) {
                return status;
            }
        }
        throw new IllegalArgumentException("Invalid room status code: " + code);
    }
}

File: front/src/constants/roomStatus.ts
import { RoomStatus } from '../types/room'

export interface RoomStatusConfig {
  label: string
  color: string
  bgColor: string
  icon: string
}

export const ROOM_STATUS_CONFIG: Record<RoomStatus, RoomStatusConfig> = {
  [RoomStatus.CLEAN]: {
    label: '空净',
    color: '#16a34a',
    bgColor: '#dcfce7',
    icon: 'check_circle',
  },
  [RoomStatus.RESERVED]: {
    label: '预留',
    color: '#f59e0b',
    bgColor: '#fef3c7',
    icon: 'bookmark',
  },
  [RoomStatus.BOOKED]: {
    label: '预订',
    color: '#3b82f6',
    bgColor: '#dbeafe',
    icon: 'event',
  },
  [RoomStatus.OCCUPIED]: {
    label: '在住',
    color: '#7c3aed',
    bgColor: '#ede9fe',
    icon: 'person',
  },
  [RoomStatus.DIRTY]: {
    label: '脏房',
    color: '#dc2626',
    bgColor: '#fee2e2',
    icon: 'cleaning_services',
  },
  [RoomStatus.OUT_OF_SERVICE]: {
    label: '停售',
    color: '#6b7280',
    bgColor: '#f3f4f6',
    icon: 'block',
  },
}

export const ROOM_STATUS_TRANSITIONS: Record<RoomStatus, RoomStatus[]> = {
  [RoomStatus.CLEAN]: [
    RoomStatus.RESERVED,
    RoomStatus.BOOKED,
    RoomStatus.OCCUPIED,
    RoomStatus.OUT_OF_SERVICE,
  ],
  [RoomStatus.RESERVED]: [RoomStatus.CLEAN, RoomStatus.OUT_OF_SERVICE],
  [RoomStatus.BOOKED]: [RoomStatus.OCCUPIED, RoomStatus.CLEAN],
  [RoomStatus.OCCUPIED]: [RoomStatus.DIRTY],
  [RoomStatus.DIRTY]: [RoomStatus.CLEAN],
  [RoomStatus.OUT_OF_SERVICE]: [RoomStatus.CLEAN],
}
